### ISSUE TITLE
Server chooses cipher suite not matching its certificate

### ISSUE THREAD

**Issue by lukaslihotzki** [pion/dtls#398]
If the server only has either ECDSA or RSA certificates, it needs to choose the corresponding cipher suite, even if the client announces a preference for a cipher suite of the other type.

How to reproduce:
1. Run the server (master branch): `go run examples/listen/selfsign/main.go`. It will generate a self-signed ECDSA certificate.
2. Connect with OpenSSL s_client, without client certificates, with preference for ECDSA (works):
```openssl s_client -dtls1_2 -connect 127.0.0.1:4444 -debug -cipher aECDSA:aRSA```
3. Do the same, but with preference for RSA (fails):
```openssl s_client -dtls1_2 -connect 127.0.0.1:4444 -debug -cipher aRSA:aECDSA```

The fail makes s_client send a fatal alert and exit immediately printing:
```
140576763131264:error:1416F17F:SSL routines:tls_process_server_certificate:wrong certificate type:ssl/statem/statem_clnt.c:1958:
```
The go server then exits too because of the fatal alert:
```
net.Error: handshake error: alert: Alert Fatal: IllegalParameter
panic: handshake error: alert: Alert Fatal: IllegalParameter
```

This issue was mentioned in https://github.com/pion/dtls/pull/350#issuecomment-798403320.

**Comment by lukaslihotzki** [comment]
It seems like certificate selection is just by server name, not by certificate type. So for each server name, the server supports either ECDSA or RSA and it should restrict the cipher suites accordingly.

**Comment by jfdive** [comment]
Had the same issue (CA issued RSA certs & keys) but client was always proposing ECDSA meaning failure to connect. the associated PR fixes the problem.

**Comment by Sean-Der** [comment]
Fixed by b8f72f3774497508fd9bc960a676dc395c572416

Thank you @lukaslihotzki !
